docs/source/data/model.md

```diff
@@ -0,0 +1,41 @@
+# Model Data Type Documentation
+
+Definces a custom data type called `ModelData` in AiiDA, which is a subclass of the `SinglefileData` type.`ModelData` is used to handle model files and provides functionalities for handling local files and downloading files from URLs.
+Additional features compared to `SinglefileData`:
+ - it can take a non absolute path as an argument (*change or leave? also in this case it has to be a tring but this can also be changed to take a path*)
+ - if given a URL it will download the file, save it in a folder of choice (default = ./cache/mlips, if given an architecture it will create a subfolder with that name), and save the file as AiiDA data type. If the file is downloaded twice it will be canceled if there are duplicates in the same folder, however this is not related with AiiDA caching for its own database. For that, enable the caching for the ModelData class, or the whole aiida-mlip plugin (https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/provenance/caching.html)
+ - *others to be added? e.g. read architecture from file itself if possible?*
+
+## Functions and Classes
+
+### `calculate_hash(file)`
```

Comment:
Should this not be coming from `Sphinx` or similar rather than being handwritten?